File: src/main/java/utility/game/repository/item/ItemCharacteristic.java
package utility.game.repository.item;

import java.util.HashMap;
import java.util.Map;

/**
 * @author Tyluur <itstyluur@icloud.com>
 * @since 9/1/2017
 */
public final class ItemCharacteristic {
	
	/**
	 * The map of bonuses
	 */
	private final Map<Integer, int[]> bonusMap = new HashMap<>();
	
	/**
	 * The map of examines
	 */
	private final Map<Integer, String> examineMap = new HashMap<>();
	
	/**
	 * Adds the bonuses of an item
	 */
	public void addBonuses(int itemId, int[] bonuses) {
		bonusMap.put(itemId, bonuses);
	}
	
	/**
	 * Gets the bonuses of an item
	 */
	public int[] getBonuses(int itemId) {
		return bonusMap.get(itemId);
	}
	
	/**
	 * Adds the examine of an item
	 */
	public void addExamine(int itemId, String examine) {
		examineMap.put(itemId, examine);
	}
	
	/**
	 * Gets the examine of an item
	 */
	public String getExamine(int itemId) {
		return examineMap.get(itemId);
	}
	
	@Override
	public String toString() {
		return "ItemCharacteristic{" + "bonusMap=" + bonusMap.size() + ", examineMap=" + examineMap.size() + '}';
	}
}

File: src/main/kotlin/org/redrune/utility/game/repository/item/ItemCharacteristic.kt
package org.redrune.utility.game.repository.item

/**
 * @author Tyluur <itstyluur@icloud.com>
 * @since 9/1/2017
 */
class ItemCharacteristic {

    /**
     * The map of bonuses
     */
    private val bonusMap: MutableMap<Int, IntArray> = HashMap()

    /**
     * The map of examines
     */
    private val examineMap: MutableMap<Int, String> = HashMap()

    /**
     * Adds the bonuses of an item
     */
    fun addBonuses(itemId: Int, bonuses: IntArray) {
        bonusMap[itemId] = bonuses
    }

    /**
     * Gets the bonuses of an item
     */
    fun getBonuses(itemId: Int): IntArray? {
        return bonusMap[itemId]
    }

    /**
     * Adds the examine of an item
     */
    fun addExamine(itemId: Int, examine: String) {
        examineMap[itemId] = examine
    }

    /**
     * Gets the examine of an item
     */
    fun getExamine(itemId: Int): String? {
        return examineMap[itemId]
    }

    override fun toString(): String {
        return "ItemCharacteristic{" + "bonusMap=" + bonusMap.size + ", examineMap=" + examineMap.size + '}'
    }
}